be/src/vec/exec/vparquet_scanner.cpp

```diff
@@ -0,0 +1,307 @@
+        auto pt = arrow_type_to_primitive_type(array->type()->id());
+        if (pt == INVALID_TYPE) {
+            return Status::NotSupported(fmt::format(
+                "Not support arrow type:{}", array->type()->name()));
+        }
+        auto is_nullable = true;
+        // let src column be nullable for simplify converting
+        DataTypePtr data_type = DataTypeFactory::instance().create_data_type(pt, is_nullable);
+        MutableColumnPtr data_column = data_type->create_column();
+        block->insert(ColumnWithTypeAndName(std::move(data_column), data_type, slot_desc->col_name()));
+    }
+    return Status::OK();
+}
+
+Status VParquetScanner::get_next(std::vector<MutableColumnPtr>& columns, bool* eof) {
+    // overall of type converting: 
+    // arrow type ==arrow_column_to_doris_column==> primitive type(PT0) ==cast_src_block==>
+    // primitive type(PT1) ==materialize_block==> dest primitive type
+    SCOPED_TIMER(_read_timer);
+    // init arrow batch
+    {
+        Status st = init_arrow_batch_if_necessary();
+        if (!st.ok()) {
+            if (!st.is_end_of_file()) {
+                return st;
+            }
+            return Status::OK();
+        }
+    }
+    Block src_block;
+    RETURN_IF_ERROR(init_src_block(&src_block));
+    // convert arrow batch to block until reach the batch_size
+    while (!_scanner_eof) {
+        // cast arrow type to PT0 and append it to src block
+        // for example: arrow::Type::INT16 => TYPE_SMALLINT
+        RETURN_IF_ERROR(append_batch_to_src_block(&src_block));
+        // finalize the src block if full
+        if (src_block.rows() >= _state->batch_size()) {
+            break;
+        }
+        auto status = next_arrow_batch();
+        // if ok, append the batch to the src columns
+        if (status.ok()) {
+            continue;
+        }
+        // return error if not EOF
+        if (!status.is_end_of_file()) {
+            return status;
+        }
+        // if src block is not empty, then finalize the block
+        if (src_block.rows() > 0) {
+            break;
+        }
+        _cur_file_eof = true;
+        RETURN_IF_ERROR(next_arrow_batch());
+        // there may be different arrow file, so reinit block here
+        RETURN_IF_ERROR(init_src_block(&src_block));
+    }
+    COUNTER_UPDATE(_rows_read_counter, src_block.rows());
+    SCOPED_TIMER(_materialize_timer);
+    // cast PT0 => PT1
+    // for example: TYPE_SMALLINT => TYPE_VARCHAR
+    RETURN_IF_ERROR(cast_src_block(&src_block));
+    // range of current file
+    fill_columns_from_path(&src_block);
+    RETURN_IF_ERROR(eval_conjunts(&src_block));
+    // materialize, src block => dest columns
+    RETURN_IF_ERROR(materialize_block(&src_block, columns));
+    *eof = _scanner_eof;
+    return Status::OK();
+}
+
+// eval conjuncts, for example: t1 > 1
+Status VParquetScanner::eval_conjunts(Block* block) {
+    for (auto& vctx : _pre_filter_vctxs) {
```

Comment:
may should change `_pre_filter_vctxs` to a tree to avoid the each time of copy operation